lib/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -306,13 +318,23 @@ impl<'a> FuncGen<'a> {
         for i in begin..end {
             self.trap_table.offset_to_code.insert(i, code);
         }
+        self.instructions_address_map.push(InstructionAddressMap {
+            srcloc: SourceLoc::new(self.src_loc),
+            code_offset: begin,
+            code_len: end - begin,
+        });
         ret
     }
 
     /// Marks one address as trappable with trap code `code`.
     fn mark_address_with_trap_code(&mut self, code: TrapCode) {
         let offset = self.assembler.get_offset().0;
         self.trap_table.offset_to_code.insert(offset, code);
+        self.instructions_address_map.push(InstructionAddressMap {
+            srcloc: SourceLoc::new(self.src_loc),
+            code_offset: offset,
+            code_len: 2, // TODO: Check this length
```

Comment:
I replicated cranelift behavior

Reply:
Hold on though. Singlepass isn't emitting the same instructions that cranelift is emitting. This is the length of the trap in Cranelift.

Ah! `ud2` is two bytes, not one! For some reason I thought `ud` and `ud2` were 1-byte instructions. OK, using 2 makes more sense because singlepass uses `ud2` nearly all the time for its traps.

_Nearly_. `emit_relaxed_xdiv` emits either a DIV or IDIV instruction. When `sz == Size::S64` then it's a 3-byte encoding (`48 f7 f8 idivq %rax`) otherwise it's a 2-byte encoding (such as `f7 f8 idiv %eax`).

Reply:
Should I change to three bytes then?

Reply:
Use `if sz == Size::S64 { 3 } else { 2 }` for the instances in `emit_relaxed_xdiv` yes.